### PR TITLE
second assignment

### PR DESCRIPTION
**Pull request by roelof1967**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -2,36 +2,28 @@
...
-        }
-    });
-
-    server.start(function (err) {
-
-        Hoek.assert(!err, err);
-        console.log('Server started at: ' + server.info.uri);
-    });
+  
+  var server = new Hapi.Server();
```
Comment (AdriVanHoudt) [inline lib/index.js]: where did your indentation go?

File: lib/index.js [hunk 2]
```diff
@@ -2,36 +2,28 @@
...
+  server.connection({ port: 3000 });
+
+  server.register({register: require('./version')}, function (err) {
+    
+    if (err) {
+        console.error('Failed to load plugin:', err);
```
Comment (AdriVanHoudt) [inline lib/index.js]: Use the same error handling here as in `server.start`

File: lib/version.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+var internals = {
+  package: require('../package.json') 
+}; 
+
+exports.register = function (server, options, next) {
+  
+  server.route({
+
```
Comment (AdriVanHoudt) [inline lib/version.js]: No newline required here

File: lib/version.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+var internals = {
+    package: require('../package.json') 
+}; 
+
+exports.register = function (server, options, next) {
+    server.route({
```
Comment (AdriVanHoudt) [inline lib/version.js]: newline after function

File: lib/version.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        config: {
+            description: 'Returns the version of the server',
+            handler: function (request, reply) {
+                
+                return reply({ version: internals.package.version });
+            }
+        }
+    });
+    
+  next();
```
Comment (AdriVanHoudt) [inline lib/version.js]: `return next()` is more readable

File: lib/version.js (hunk 4 above)
Comment (roelof1967) [inline lib/version.js]: Now im confused. At the last remark before this one you said no new line required here and now you said you want a newline. Can you explain this ?

Comment (AdriVanHoudt) [inline lib/version.js]: previous was no newline after `route({` this one is about a newline after `function () {`

File: lib/index.js [hunk 6]
```diff
@@ -2,36 +2,26 @@
...
-        }
-    });
-
-    server.start(function (err) {
-
-        Hoek.assert(!err, err);
-        console.log('Server started at: ' + server.info.uri);
-    });
+    
+    server.register({register: require('./version')}, function (err) {
```
Comment (hueniverse) [inline lib/index.js]: No need to wrap the plugin in `{register}`, just pass it directly. Also, move the `require()` up like any other module.

File: package.json [hunk 7]
```diff
@@ -18,8 +18,8 @@
   },
   "homepage": "https://github.com/hueniverse/hueniversity",
   "dependencies": {
-    "hapi": "8.x.x",
-    "hoek": "2.x.x"
+    "hapi": "^8.4.0",
+    "hoek": "^2.11.1"
```
Comment (hueniverse) [inline package.json]: Why change this?!

**Pull request closed without merging**